src/math/matrix.c

```diff
@@ -25,56 +25,66 @@
  *   -EINVAL if input dimensions do not allow for multiplication.
  *   -ERANGE if the shift operation might cause integer overflow.
  */
-int mat_multiply(struct mat_matrix_16b *a, struct mat_matrix_16b *b, struct mat_matrix_16b *c)
+int mat_multiply(struct mat_matrix_16b *a, struct mat_matrix_16b *b,
+		 struct mat_matrix_16b *c)
```

Comment:
please, split this commit: a single commit per optimisation and separate commits with any variable renaming, comments, and similar cosmetic changes, unrelated to actual optimisations. And if you claim "36.31% reduction in memory usage" I'd like to understand where it comes from, so far I don't.

Reply:
Sure, I'II make.